### PR TITLE
Add Models to Bookmarks app

### PR DESCRIPTION
**Pull request by pavan-testpress**
- Add Bookmarksapp to Installed apps
- Add Bookmarks Model
- Add Folders Model
- Add django_utils Pipfile
- Migrate the Models
- Remove Unused imports to follow flake8 guidelines
- Add Bookmarks, Folders Models to Admin site

### CODE REVIEW

File: DjangoProject/bookmarksapp/admin.py [hunk 1]
```diff
@@ -1,3 +1,8 @@
 from django.contrib import admin
 
+from .models import Bookmarks, Folders
+
 # Register your models here.
```
Comment (waranlogesh) [inline DjangoProject/bookmarksapp/admin.py]: Remove the auto-generated comments

File: DjangoProject/bookmarksapp/models.py [hunk 2]
```diff
@@ -1,3 +1,30 @@
 from django.db import models
+from django.utils.text import slugify
+
+from model_utils.models import TimeStampedModel
+
+from authenticationapp.models import MyUser
 
 # Create your models here.
```
Comment (waranlogesh) [inline DjangoProject/bookmarksapp/models.py]: Remove it

File: DjangoProject/bookmarksapp/models.py [hunk 3]
```diff
@@ -1,3 +1,30 @@
...
+
+
+class Folders(TimeStampedModel):
+    name = models.CharField(max_length=50, unique=True)
```
Comment (waranlogesh) [inline DjangoProject/bookmarksapp/models.py]: `unique = True`? So what if another user whats to create the same folder name.

Consider `user1` has created a folder with the name `google` and `user2` tries to create a folder with the same name `google` and your design won't allow that.

Ideally, it should be allowed and uniqueness for folder name should be check with respect to the user.

File: DjangoProject/bookmarksapp/models.py [hunk 4]
```diff
@@ -1,3 +1,30 @@
...
+    def save(self, *args, **kwargs):
+        self.slug = slugify(self.name)
+        super(Folders, self).save(*args, **kwargs)
+
+
+class Bookmarks(TimeStampedModel):
+    folder = models.ForeignKey(Folders, on_delete=models.SET_NULL, related_name='folder', null=True)
+    name = models.CharField(max_length=50, unique=True)
+    url = models.URLField(max_length=200, unique=True)
+    user = models.ForeignKey(MyUser, on_delete=models.CASCADE, default=1)
```
Comment (waranlogesh) [inline DjangoProject/bookmarksapp/models.py]: Why do u have a user field here? Field name should be verbose. If it's for storing the user who created the bookmark it should be `created_by`. Why is `default=1`?

**Review by waranlogesh — CHANGES_REQUESTED** [review]

File: DjangoProject/bookmarksapp/admin.py (hunk 1 above)
Comment (pavan-testpress) [reply DjangoProject/bookmarksapp/admin.py]: Removed the auto-generated comments

File: DjangoProject/bookmarksapp/models.py (hunk 2 above)
Comment (pavan-testpress) [reply DjangoProject/bookmarksapp/models.py]: Removed the auto-generated comments

File: DjangoProject/bookmarksapp/models.py (hunk 3 above)
Comment (pavan-testpress) [reply DjangoProject/bookmarksapp/models.py]: I removed unique constraint. will check in views while saving.

File: DjangoProject/bookmarksapp/models.py (hunk 4 above)
Comment (pavan-testpress) [reply DjangoProject/bookmarksapp/models.py]: - changed user to created_by.
- added null=True
- remove default = 1

**Review by waranlogesh — APPROVED** [review]

**Pull request merged**